python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,1138 @@
+                            if isinstance(self._op_inputs_r[(op_name, operator)][cnt], str):
+                                if "call/var" in self._op_inputs_r[(op_name, operator)][cnt]:
+                                    self._op_inputs_r[(op_name, operator)][cnt] = \
+                                        self._relay_map[self._nid_to_node_name[i.debugName()]]
+                                    break
+
+                call = _convert_map[operator](self._op_inputs_r[(op_name, operator)],
+                                              self._op_inputs_types[(op_name, operator)])
+
+                self._relay_map[nid] = call
+                self._nid_to_node_name[op_name] = nid
+                nid = nid+1
+
+        outputs = []
+
+        for i in range(nid):
+            output = self._relay_map[i]
+            outputs.append(output)
+
+        if len(outputs) == 1:
+            body = outputs[0]
+        else:
+            body = outputs[-1]
+
+        func = tvm.relay.Function(_analysis.free_vars(body), body)
+
+        param = {k: tvm.nd.array(v) for k, v in self._param_tensors.items()}
+
+        return  _module.Module.from_expr(func), param
+
+    def _parse_inputs(self):
+        """ Map inputs to parser and inputs to graph. """
+        # Get names and objects of inputs for IR
+        ir_names = [i.debugName() for i in self._trace.graph.inputs()]
+        ir_inputs = [i for i in self._trace.graph.inputs()]
+
+        # Create corresponding shape and add to input
+        for input_name, ir_input in zip(self._input_shapes, ir_inputs[1:]):
+            input_shape = self._input_shapes[input_name]
+            ir_input.setDebugName(input_name)
+
+            self._inputs_r[input_name] = _expr.var(input_name,
+                                                   shape=self._input_shapes[input_name],
+                                                   dtype=_convert_data_type(
+                                                       self._input_types[input_name]))
+            self._fn_param.append(_expr.var(input_name,
+                                            shape=self._input_shapes[input_name],
+                                            dtype=_convert_data_type(
+                                                self._input_types[input_name])))
+
+        # Add self (first input of a PyTorch graph) to inputs
+        input_shape = [3]
+        tensor = tvm.nd.array(np.zeros(input_shape).astype(np.float32))
+        input_name = ir_names[0]
+        self._inputs_r[input_name] = tensor
+
+    def _parse_params(self):
+        """ Map state dictionary values to corresponding prim::GetAttr op node. """
+        # Grab weights, biases, etc. from graph
+        state_dict = self._trace.state_dict()
+        param_names = []
+        for key, value in state_dict.items():
+            param_str = str(key)
+            param_name = param_str.split('.')[-1]
```

Comment:
removing everything except the last looks suspicious. With the latest PyTorch, if I run this on a sequence of two conv + bn + relu blocks below, I get pairs of (key, param_name) at the bottom.
```
class ConvBNReLU(nn.Sequential):
    def __init__(self, in_planes, out_planes, kernel_size=3, stride=1, groups=1):
        padding = (kernel_size - 1) // 2
        super().__init__(
            nn.Conv2d(in_planes, out_planes, kernel_size, stride, padding, groups=groups, bias=False),
            nn.BatchNorm2d(out_planes, momentum=0.1),
            # Replace with ReLU
            nn.ReLU(inplace=False)
        )
```

```
(key, param_name): (0.0.weight, weight)
(key, param_name): (0.1.bias, bias)
(key, param_name): (0.1.weight, weight)
(key, param_name): (0.1.running_mean, running_mean)
(key, param_name): (0.1.running_var, running_var)
(key, param_name): (0.1.num_batches_tracked, num_batches_tracked)
(key, param_name): (1.0.weight, weight)
(key, param_name): (1.1.bias, bias)
(key, param_name): (1.1.weight, weight)
(key, param_name): (1.1.running_mean, running_mean)
(key, param_name): (1.1.running_var, running_var)
(key, param_name): (1.1.num_batches_tracked, num_batches_tracked)
```

Reply:
if this is indeed an intended behavior, I suggest updating param_name from list to set.